cmd/influxd/inspect/dump_tsm_wal.go

```diff
@@ -0,0 +1,50 @@
+package inspect
+
+import (
+	"github.com/influxdata/influxdb/kit/errors"
+	"github.com/influxdata/influxdb/storage/wal"
+	"github.com/spf13/cobra"
+	"os"
+)
+
+var dumpTSMWALFlags = struct {
+	findDuplicates bool
+}{}
+
+func NewDumpTSMWALCommand() *cobra.Command {
+	dumpTSMWALCommand := &cobra.Command{
+		Use:   "dumptsmwal",
+		Short: "Dump TSM data from WAL files",
+		Long: `
+This tool dumps data from WAL files for debugging purposes. Given a list of .wal files,
+the tool will parse and print out the entries in each file. For each file, the following is printed:
+	* The file name
+	* for each entry,
+		* The type of the entry (either [write] or [delete-bucket-range]);
+		* The formatted entry contents
+If the tool is run with --find-duplicates=true, A list of all keys with duplicate or out of order timestamps will 
+be printed.
+`,
+		RunE: inspectDumpTSMWAL,
+	}
+
+	dumpTSMWALCommand.Flags().BoolVarP(&dumpTSMWALFlags.findDuplicates, "find-duplicates", "", false, "report keys with out of order points")
```

Comment:
Looks like this flag means "dupe check, not dump". It's almost a separate sub-command. Please make the flag description clearer about this either-or behavior.